src/shims/posix/linux/foreign_items.rs

```diff
@@ -183,7 +183,10 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                     id if id == sys_futex => {
                         futex(this, args, dest)?;
                     }
-                    id => throw_unsup_format!("Miri does not support syscall ID {}", id),
+                    id => {
+                        this.handle_unsupported(format!("can't execute syscall with ID {}", id))?;
+                        return Ok(EmulateByNameResult::NotSupported);
```

Comment:
Should this be `EmulateByNameResult::AlreadyJumped`?

Reply:
Hmmm, I think you are right. This is one of the scenarios I did not test and may have been a merge error.

Reply:
Ah right, `NotSupported` means "look up exported function symbol".

I wonder if we should look up the exported symbol *first*, and only then run our own shim?

Reply:
> I wonder if we should look up the exported symbol _first_, and only then run our own shim? Then we might not even need the `NotSupported` variant.

I think the variant is still needed -- the exported symbol will *only* be run if no shim implementation can be found, and `NotSupported` is required to represent that.